### ISSUE TITLE
Danishbits not getting results or all results, might be missing som tv & movie categories

### ISSUE THREAD

**Issue by estavez** [Jackett/Jackett#2233]
Jackett version:  
0.8.489.0

I was the first to help you out with adding the Hounddawgs and Danishbits tracker case here with another guy I don't know.

When using Sonarr or Radarr I've been having problems with Danisbits over quite some time now with getting results or all results. I've tried reaching out to Danishbits them selves, but they say that everything is fine at there end. And this is sad because it is a really good tracker and we need to support it otherwise it might go away.

I'm also using Hounddawgs with Jackett and this just seems to work every time, but it is good to have them both as Danishbits often have new releases or pre-releases first.

I think this may have to do with Danishbits tracker at Jackett end is missing most categories.

If you see the Houndawgs categories they are complete:
5000 | TV
5030 | TV/SD
5040 | TV/HD
2050 | Movies/3D
100068 | 3D
4070 | PC/Phone-Android
100080 | Appz / Android
4010 | PC/0day
100086 | Appz / Div
4060 | PC/Phone-IOS
100071 | Appz / iOS
4030 | PC/Mac
100070 | Appz / Mac
100069 | Appz / PC
3030 | Audio/Audiobook
100072 | Audio Books
2060 | Movies/BluRay
100082 | BluRay/REMUX
8000 | Books
100078 | Books
7000 | Other
100087 | Cover
2070 | Movies/DVD
100090 | DK DVDr
2040 | Movies/HD
100089 | DK HD
2030 | Movies/SD
100091 | DK SD
100092 | DK TV HD
100093 | DK TV SD
100083 | ELearning
2000 | Movies
100084 | Film Boxset
100081 | Film CAM/TS
100060 | Film DVDr
100059 | Film HD
100073 | Film SD
2020 | Movies/Other
100077 | Film Tablet
3000 | Audio
100061 | Musik
3020 | Audio/Video
100076 | MusikVideo/Koncert
1000 | Console
100075 | Spil / Konsol
100079 | Spil / Mac
4050 | PC/Games
100064 | Spil / PC
100085 | TV Boxset
100058 | TV DVDr
100057 | TV HD
100074 | TV SD
5050 | TV/OTHER
100094 | TV Tablet
6000 | XXX
100067 | XXX

But if you look into Danishbits categories, this is only filled in Jackett with 2 categories despite that they have nearly the same amount of categories as default.

2000 | Movies
5000 | TV

We are missing the following to get this working correctly as I see it.
Blu-ray Film 
Danske Film
Dansk TV
Dansk Musik
Diverse
DVD Film
DVD Film (UNiTY)
E-bøger
Film Bokssæt
HD x264 Film
HD Film (RAPiDCOWS)
HD Film (UNiTY)
HD TV (DBRETAiL)
Lydbøger
Mac Apps
Mobil Apps
Musik
Musikvideoer
Nintendo Spil
Nordiske DVD Film
PC Apps
PC Spil
PlayStation Spil
SD Film
Tablet Film/TV
TV
TV Bokssæt
TV (Substance)
Xbox Spil
XXX 

Not all might be needed but all categories for TV and Movies (in Danish its called "Film").

I already tried to put more categories in Sonar/Radarr under Indexer categories, but it does not help this out so it seems to come down to the fact that Danishbits tracker in Jackett is missing the main categories like HD TV and HD Films categories. 

Can you help getting these added to the next release? I cannot find the old post where all the category numbers was, maybe you can help out? Not sure how to get these.

And for last but at least I would like to say that you are doing a fantastic work with Jackett, without this we would not be able to use our Private trackers today. Thank you very much. 
BTW, is there a place where I can donate you for all you Jackett work?

**Comment by kaso17** [comment]
earlier danishbits was scraping the normal browser search. But they updated their rules and forbid this. Since then jackett is using their couchpotato API. Unfortunately it's very buggy/missing data.

I've sent them these staff PMs.
On the first one I've got a "we'll look into this" (but nothing changed), the other ones didn't get answered at all.
My ticket was auto closed after 21 days...
As I don't use the tracker myself I don't care very much, if you've a better contact feel free to remind them again.

```
Hello, I noticed that the couchpotato API returns the wrong time format.
According to https://github.com/CouchPotato/CouchPotatoServer/wiki/Couchpotato-torrent-provider
publish_date should be in UTC ISO 8601 date time format.

Additionally it seems like "type" attribute can be only "movie" or "tv". E.g. this example returns type tv but it's a game:
couchpotato.php?search=Ashes.of.the.Singularity&user=XXX&passkey=XXX

Additionally according to the API specification "show" should be used for TV Series.

Besides that I'm wondering if there's a way to get the most recent releases (instead of using the RSS feed).
unfortunately searching for the empty string couchpotato.php?search=&user=XXX&passkey=XXX only results in an API error.
```

```
I'm updating the Jackett integration (It was using screen scraping before).
Due to the new API several features were lost.
- It doesn't provide the number of files
- It doesn't provide the number of downloads
- It doesn't provide the cover image
- It's not possible to see if a torrent has double upload
- Original (detailed) Categories are lost
- It's not possible to filter by category
- It only returns up to 20 results (no support for more results/multiple pages)
- It returns the oldest 20 torrents

It would be great if you could enhance it accordingly.
Especially the last point is very annoying. Is there any way to change the sort order (Get the most recent instead of the oldest results)?
```

```
There's also a problem with the category mapping of NORDIC TV releases: 
See this example
couchpotato.php?search=room%25104%25nordic&passkey=XXX&user=XXX
In all cases type is movie but it should be tv/show.
```

**Comment by estavez** [comment]
Thanks @kaso17 for the quick response and the pre-writings you already have tried to reach out to Danishbits on. Sorry to hear they did not listen or got back to you.

I can see that I haven't updated my settings to use the Couchpotato api, but I've tried and cannot get this to work with Connection succeed? Is there some special settings I need to set, because if I change the Torsnab URL to potato feed URL it does not work at all, could you point me in the right direction on these settings for Sonarr and Radarr?

I could try to write to Danishbits to see if I can get through to them, maybe my words would mean more as I'm an heavy user uploader at there site and I think they would not like to have me gone from the site.

But after this I was curious to find out if the Danishbits missing results, where a general problem so I tested it out in the forked version named "Cardigann" as I have tried this in an earlier state because our hosting provider was threatened with ending the "Mono engine" support on there systems because on Linux boxes they took up quite a lot of instances and unessesary resources in earlier releases of Mono engine. But after this got better from newer Mono releases they let us all have the Mono support so we could still be using Jackett, Sonarr and Radarr etc.

Point being, that the Danishbit is working as it should on Sonarr fetching all the right results every time if I connect it to Cardigann, so it might be easier for all to just look into there coding if that is an option? 

Problems with Cardigann for us is that there are no support for Radarr or our other tracker Hounddawgs at the moment so I did not get the chance to test this out really and at that earlier time I tried to reach out to them and got an answer that the project was set on hold for unlimited time a year or two back. But I see they are working on it again now.

**Comment by kaso17** [comment]
This has nothing to do if you use the couchpotato or torznab API endpoint of jackett. The problems exist in both cases and is caused by the mentioned limitations/bugs of the danishbits API.

cardigann is still using the screen scraping approach which is against the danishbits tracker rules.
If we revert the danishbit change in jackett it will work perfectly fine again unless you're a heavy API user. If I remember correctly some users got IP banned for sending to many queries to the torrents.php page (Issue #1715 and related). That's why we changed it.
I suggest you try to contact them pointing them to this issue or my danishbits helpdesk ticket. The conversation ID is 8280797
It shouldn't be hard for them make the necessary changes to their API.

**Comment by supremacy2k** [comment]
I have created a forum post, referring to your conversation id.
I hope they fix it, as it's frustrating as **** :/

**Comment by supremacy2k** [comment]
It should now be fixed.
At least the series i have tested on.

**Comment by kaso17** [comment]
Great. currently the API is broken (500 Internal Server Error), hopefully because he's working on the other buggy/missing features :)

**Comment by supremacy2k** [comment]
Works fine here.
Remember to change the URI from http to https :)

**Comment by kaso17** [comment]
ok it's fixed but no changes besides the category mapping.
Maybe you can ask him about them?

**Comment by supremacy2k** [comment]
I'm not sure, they actually did anything else, other than changing the tv section to be listed as tv and not movie.

**Comment by supremacy2k** [comment]
Update:

Apps, Music, Games etc, is now listed under "blandet" (meaning: Mixed, in english)
all tv shows should now be correctly listed under "TV"
I asked about, changing the sort order.

Also handed over at link to this issue.

**Comment by alxlaxv** [comment]
Last updates fixed your issue @supremacy2k ?

**Comment by InfAmoUsaeble** [comment]
Still not working for me on Version 0.8.815.0

**Comment by supremacy2k** [comment]
It's working like a charm for me.
Jackett reports no issues, but Sonarr fails in testing, but it still works.

**Comment by InfAmoUsaeble** [comment]
I can test without a problem on either indexer, but results are 3-4 years old at best, when searching.

**Comment by kaso17** [comment]
@InfAmoUsaeble that's a danishbits problem, their API doesn't allow sorting and by defaults shows the oldest results first. You'll have to use more specific search terms or contact the admins to make them fix the API.

**Comment by supremacy2k** [comment]
I'm still getting

"Invalid request Validation failed: 
 -- No results were returned from your indexer, please check your settings."

When trying to add danishbits to sonarr.
Even with trace logging enabled, nothing more shows up in logs. :/

Currently I'm not able to add danishbits from jackett to sonarr.

**Comment by ngosang** [comment]
@RoloSoze I looked at the code and I can't do anything to fix the open issues for Danishbits. The API is garbage. If we scrape the website (HTML) instead of using the API I can fill all problems, but according to https://github.com/Jackett/Jackett/issues/2233#issuecomment-348991122 it's forbidden.
Maybe you can talk with the admin to abolish the prohibition. We can limit the requests/second or something if requested.

**Comment by RoloSoze** [comment]
> 
> 
> @RoloSoze I looked at the code and I can't do anything to fix the open issues for Danishbits. The API is garbage. If we scrape the website (HTML) instead of using the API I can fix all problems, but according to [#2233 (comment)](https://github.com/Jackett/Jackett/issues/2233#issuecomment-348991122) it's forbidden.
> Maybe you can talk with the admin to abolish the prohibition. We can limit the requests/second or something if requested.

@ngosang This was also my understanding; that action needs to be taken by the DB staff in order to solve these issues. As I detailed in https://github.com/Jackett/Jackett/issues/3467#issuecomment-612270002, I'm personally not going to restart the conversation with the DB staff since it's not a priority tracker for me,  

However, @supremacy2k @estavez have both been involved in some of the DanishBits issues raised in Jackett (this one, #2657, #3467), and I believe at least one of them have spoken to the staff a long time ago. Maybe they can assist with restarting the conversation? It might work out this time.

**Comment by ngosang** [comment]
If they don't answer in few weeks we can merge all Danishbits issues in one to clean up the list a bit.

**Comment by ngosang** [comment]
@RoloSoze could you merge all danishbits issues into one? there are at least 3

**Comment by RoloSoze** [comment]
@ngosang I'm not sure it's a good idea to merge all 3 issues in one because each has its own discussion thread, so even if I put them all together under one master issue, it'll keep referencing the existing 3. Also, I believe solving the issues requires contact with DB's staff, and as I mentioned here https://github.com/Jackett/Jackett/issues/2233#issuecomment-612507066, contact has been established in the past, but now it needs to be followed up, and no one has stepped up to do this, so what's your plan?

**Comment by ngosang** [comment]
no plan for this. it is possible that in several years the site will disappear and the problem will solve itself.

**Comment by RoloSoze** [comment]
Calling on @supremacy2k @estavez one last time to see if either of them is willing to reestablish contact with DanishBits staff in order to help resolve its issues. If they're willing to do that, it'll be worth consolidating all 3 issues into one.

**Comment by garfield69** [comment]
closed due to #9949
